pkg/stdlib/sql.go

```diff
@@ -0,0 +1,386 @@
+	return math.MaxInt64, false
+}
+
+// ColumnTypePrecisionScale should return the precision and scale for decimal
+// types. If not applicable, ok should be false.
+func (r *Rows) ColumnTypePrecisionScale(index int) (precision, scale int64, ok bool) {
+	return 0, 0, false
+}
+
+// ColumnTypeScanType returns the value type that can be used to scan types into.
+func (r *Rows) ColumnTypeScanType(index int) reflect.Type {
+	return nil
+}
+
+func (r *Rows) Close() error {
+	return r.conn.Close()
+}
+
+func (r *Rows) Next(dest []driver.Value) error {
+	if r.index >= uint64(len(r.rows)) {
+		return io.EOF
+	}
+
+	row := r.rows[r.index]
+
+	for idx, val := range row.Values {
+		dest[idx] = schema.RenderValueAsByte(val.Value)
+	}
+
+	r.index++
+	return nil
+}
+
+func namedValuesToSqlMap(argsV []driver.NamedValue) (map[string]interface{}, error) {
+	args := make([]interface{}, 0, len(argsV))
+	for _, v := range argsV {
+		if v.Value != nil {
+			args = append(args, v.Value.(interface{}))
+		} else {
+			args = append(args, nil)
+		}
+	}
+
+	args, err := convertDriverValuers(args)
+	if err != nil {
+		return nil, err
+	}
+
+	vals := make(map[string]interface{})
+
+	for id, nv := range args {
+		key := "param" + strconv.Itoa(id+1)
+		switch args[id].(type) {
+		case string:
+			vals[key] = nv
+		case time.Time:
+			vals[key] = nv.(time.Time).Unix()
+		case uint:
+			vals[key] = int64(nv.(uint))
+		case float64:
+			return nil, ErrFloatValuesNotSupported
+		case interface{}:
+			vals[key] = nv
+		default:
+			vals[key] = nv
+		}
+	}
+	return vals, nil
+}
+
+func convertDriverValuers(args []interface{}) ([]interface{}, error) {
+	for i, arg := range args {
+		switch arg := arg.(type) {
+		case driver.Valuer:
+			v, err := callValuerValue(arg)
+			if err != nil {
+				return nil, err
+			}
+			args[i] = v
+		}
+	}
+	return args, nil
+}
+
+var valuerReflectType = reflect.TypeOf((*driver.Valuer)(nil)).Elem()
+
+// callValuerValue returns vr.Value()
+// This function is mirrored in the database/sql/driver package.
+func callValuerValue(vr driver.Valuer) (v driver.Value, err error) {
+	if rv := reflect.ValueOf(vr); rv.Kind() == reflect.Ptr &&
+		rv.IsNil() &&
+		rv.Type().Elem().Implements(valuerReflectType) {
+		return nil, nil
+	}
+	return vr.Value()
+}
+
+func ParseConfig(uri string) (*client.Options, error) {
+	if uri != "" && strings.HasPrefix(uri, "immudb://") {
+		url, err := url.Parse(uri)
+		if err != nil {
+			return nil, ErrBadQueryString
+		}
+		pw, _ := url.User.Password()
+		port, _ := strconv.Atoi(url.Port())
+		cliOpts := client.DefaultOptions().
+			WithUsername(url.User.Username()).
+			WithPassword(pw).
+			WithPort(port).
+			WithAddress(url.Hostname()).
+			WithDatabase(url.Path[1:])
+
+		return cliOpts, nil
+	}
+	return nil, ErrBadQueryString
+}
+
+func GetUri(o *client.Options) (string, error) {
+	uri := strings.Join([]string{"immudb://", o.Username, ":", o.Password, "@", o.Address, ":", strconv.Itoa(o.Port), "/", o.Database}, "")
```

Comment:
Ok for the initial version but we should build it using golang's `uri.URI.String()` method to ensure everything is properly escaped etc.

That's especially important with the password that may contain special characters such as '@'.